From d4c15a3e5d98009c167a9e8deaba9bbe880591bf Mon Sep 17 00:00:00 2001
From: Timo Rothenpieler <timo@rothenpieler.org>
Date: Sun, 22 Dec 2024 03:06:17 +0100
Subject: [PATCH 09/21] UPSTREAM: avformat/flvdec: propagate
 av_packet_add_side_data failure

(cherry picked from commit 770f0a243419f14a56f1da76b6cf44157565f98e)
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 libavformat/flvdec.c | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/libavformat/flvdec.c b/libavformat/flvdec.c
index a73a6c201238..b036204cf3b7 100644
--- a/libavformat/flvdec.c
+++ b/libavformat/flvdec.c
@@ -1720,20 +1720,22 @@ retry_duration:
             ret = av_packet_add_side_data(pkt, AV_PKT_DATA_NEW_EXTRADATA,
                                           flv->new_extradata[stream_type],
                                           flv->new_extradata_size[stream_type]);
-            if (ret >= 0) {
-                flv->new_extradata[stream_type]      = NULL;
-                flv->new_extradata_size[stream_type] = 0;
-            }
+            if (ret < 0)
+                return ret;
+
+            flv->new_extradata[stream_type]      = NULL;
+            flv->new_extradata_size[stream_type] = 0;
         } else if (multitrack &&
                    flv->mt_extradata_cnt > track_idx &&
                    flv->mt_extradata[track_idx]) {
             ret = av_packet_add_side_data(pkt, AV_PKT_DATA_NEW_EXTRADATA,
                                           flv->mt_extradata[track_idx],
                                           flv->mt_extradata_sz[track_idx]);
-            if (ret >= 0) {
-                flv->mt_extradata[track_idx]      = NULL;
-                flv->mt_extradata_sz[track_idx] = 0;
-            }
+            if (ret < 0)
+                return ret;
+
+            flv->mt_extradata[track_idx]      = NULL;
+            flv->mt_extradata_sz[track_idx] = 0;
         }
         if (stream_type == FLV_STREAM_TYPE_AUDIO && !enhanced_flv &&
                         (sample_rate != flv->last_sample_rate ||
-- 
2.48.1.windows.1

